uxarray/grid/geometry.py

```diff
@@ -582,3 +599,343 @@ def _insert_pt_in_latlonbox(old_box, new_pt, is_lon_periodic=True):
+        bounds_dataarray = grid._populate_bounds(is_latlonface=True)
+
+    This will calculate and store the bounds for each face within the grid, adjusting for any special conditions such as crossing the antimeridian, and return them as a DataArray.
+    """
+    temp_latlon_array = np.full((grid.n_face, 2, 2), INT_FILL_VALUE, dtype=np.float64)
+
+    # Because Pandas.IntervalIndex does not support naming for each interval, we need to create a mapping
+    # between the intervals and the face indices
+    intervals_tuple_list = []
+    intervals_name_list = []
+
+    for face_idx, face_nodes in enumerate(grid.face_node_connectivity):
+        face_edges_connectivity_cartesian = _get_cartesian_face_edge_nodes(
+            grid.face_node_connectivity.values[face_idx],
+            grid.face_edge_connectivity.values[face_idx],
+            grid.edge_node_connectivity.values,
+            grid.node_x.values,
+            grid.node_y.values,
+            grid.node_z.values,
+        )
+
+        face_edges_connectivity_lonlat_rad = _get_lonlat_rad_face_edge_nodes(
+            grid.face_node_connectivity.values[face_idx],
+            grid.face_edge_connectivity.values[face_idx],
+            grid.edge_node_connectivity.values,
+            grid.node_lon.values,
+            grid.node_lat.values,
+        )
+
+        is_GCA_list = (
+            is_face_GCA_list[face_idx] if is_face_GCA_list is not None else None
+        )
+
+        temp_latlon_array[face_idx] = _populate_face_latlon_bound(
+            face_edges_connectivity_cartesian,
+            face_edges_connectivity_lonlat_rad,
+            is_latlonface=is_latlonface,
+            is_GCA_list=is_GCA_list,
+        )
+
+        assert temp_latlon_array[face_idx][0][0] != temp_latlon_array[face_idx][0][1]
+        assert temp_latlon_array[face_idx][1][0] != temp_latlon_array[face_idx][1][1]
+        lat_array = temp_latlon_array[face_idx][0]
+
+        # Now store the latitude intervals in the tuples
+        intervals_tuple_list.append((lat_array[0], lat_array[1]))
+        intervals_name_list.append(face_idx)
+
+    # Because Pandas.IntervalIndex does not support naming for each interval, we need to create a mapping
+    # between the intervals and the face indices
+    intervalsIndex = pd.IntervalIndex.from_tuples(intervals_tuple_list, closed="both")
+    df_intervals_map = pd.DataFrame(
+        index=intervalsIndex, data=intervals_name_list, columns=["face_id"]
+    )
+
+    return xr.DataArray(
+        temp_latlon_array,
+        dims=["n_face", "Two", "Two"],
+        attrs={
+            "cf_role": "face_latitude_bounds",
+            "_FillValue": INT_FILL_VALUE,
+            "long_name": "Provides the latitude and longitude bounds for each face in the grid in radian",
+            "start_index": INT_DTYPE(0),
+            "latitude_intervalsIndex": intervalsIndex,
+            "latitude_intervals_name_map": df_intervals_map,
+        },
+    )
```

Comment:
Instead of returning a DataArray, can you make this populate the internal `Grid._ds` with `face_bounds`. The DataArray here looks great though! 

```Python
grid._ds['face_bounds'] = ...
```